src/libs/ReportUtils.ts

```diff
@@ -4155,9 +4155,27 @@ function doesTransactionThreadHaveViolations(report: OnyxEntry<Report>, transact
     if (report?.stateNum !== CONST.REPORT.STATE_NUM.OPEN && report?.stateNum !== CONST.REPORT.STATE_NUM.SUBMITTED) {
         return false;
     }
+
     return TransactionUtils.hasViolation(IOUTransactionID, transactionViolations);
 }
 
+/**
+ * Checks if we should display violation - we display violations when the money request has violation and it is not settled
+ */
+function shouldDisplayTransactionThreadViolations(
+    report: OnyxEntry<Report>,
+    transactionViolations: OnyxCollection<TransactionViolation[]>,
+    parentReportAction: OnyxEntry<ReportAction>,
+): boolean {
+    if (parentReportAction?.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
+        return false;
+    }
```

Comment:
Is this necessary? It's new logic that we didn't have before, and if parentReportAction is null or doesn't have originalMessage, it's the same (so long as we update to `const {IOUReportID} = parentReportAction?.originalMessage ?? {};`)